api/v1alpha1/managednotification_types.go

```diff
@@ -294,4 +293,18 @@ func (c *Conditions) SetCondition(new NotificationCondition) {
 		}
 	}
 	*c = append(*c, new)
-}
\ No newline at end of file
+}
+
+// ShouldSendResolved checks whether OA should send the SL for resolved alert
+func (m ManagedNotification) ShouldSendResolved(n string, firing bool) (bool, error) {
+	// logic for deciding whether a resolved SL should be sent goes here
+	// If no notification exists, one cannot be sent
+	t, err := m.GetNotificationForName(n)
+	if err != nil {
+		return false, err
+	}
+	if firing || len(t.ResolvedDesc) > 0 {
```

Comment:
Wouldn't this mean that it would always return true if firing is true, regardless of whether or not the resolvedDesc is empty or not? We would want it to not send a resolved SL if firing is true and resolvedDesc is empty.

Reply:
Maybe a unit test can/should be added for this function as well.

Reply:
Hi @mrbarge, I have another idea about this. Since we already have a method `CanBeSent`, I believe using one integrated method to check whether we should send SL can make it clean and tidy. And the corresponding [PR](https://github.com/openshift/ocm-agent/pull/47) was also updated. The key thing we should care about the not sending resolved alert is to update its status. So I add the logic there. Please take a look to see whether I am doing right. I have tested them. The results are the same as before.

Reply:
That's a great solution and thank you for adding the test for it as well.